types/share_merging.go

```diff
@@ -212,71 +212,65 @@ func parseMsgShares(shares [][]byte) ([]Message, error) {
 	if len(shares) == 0 {
 		return nil, nil
 	}
-
-	// set the first nid and current share
-	nid := shares[0][:consts.NamespaceSize]
-	currentShare := shares[0][consts.NamespaceSize:]
-	// find and remove the msg len delimiter
-	currentShare, msgLen, err := ParseDelimiter(currentShare)
-	if err != nil {
-		return nil, err
+	// msgs returned
+	msgs := []Message{}
+	// current msg len
+	msgLen := 0
+	// current msg
+	currentMsg := Message{}
+	// the current share contains the start of a new message
+	newMessage := true
+	// the len in bytes of the current chunk of data that will eventually become
+	// a message. This is identical to len(currentMsg.Data) + consts.MsgShareSize
+	// but we cache it here for readability
+	dataLen := 0
+
+	saveLatestMessage := func() {
```

Comment:
Probably the name `saveMessage` or `saveCurrentMessage` would be less misleading. Personally, I thought there will be multiple messages parsed and cached, but we will only save the latest one. While, if I understand right, we're reading only one message at a time.